watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Activities/MainActivity.java

```diff
@@ -49,58 +49,33 @@
 
 import org.json.JSONObject;
 
-import java.util.Arrays;
 import java.util.HashMap;
-import java.util.List;
 
 public class MainActivity extends ActionBarActivity
                           implements ActionBar.TabListener,
                                      View.OnClickListener,
-                                     ListView.OnItemClickListener,
-                                     TaskFragment.OnFragmentInteractionListener,
-                                     TaskDetailFragment.OnFragmentInteractionListener,
-                                     SiteFragment.OnFragmentInteractionListener,
-                                     UserFragment.OnFragmentInteractionListener,
-                                     AboutFragment.OnFragmentInteractionListener,
-                                     SiteListFragment.OnFragmentInteractionListener,
-                                     MiniSiteFragment.OnFragmentInteractionListener,
-                                     AddFieldReportFragment.OnFragmentInteractionListener,
-                                     TaskAbstractFragment.OnFragmentInteractionListener,
-                                     FieldReportFragment.OnFragmentInteractionListener {
+                                     ListView.OnItemClickListener {
 
     // Constants
-    public  static final String PREFERENCES = "LOGIN_PREFERENCES";
+    private static final String PREFERENCES = "LOGIN_PREFERENCES";
     private static final String TAG         = "MainActivity";
+    private static int CACHE_SIZE = 512;
 
     // Authenticating against our own server
     public String authToken;
     public String authEmail;
 
-    // For storing our credentials once we have successfully authenticated
-    SharedPreferences preferences;
-
     // Fragments
-    private TaskFragment mtaskFragment;
-    private SiteListFragment siteListFragment;
-    private FragmentManager fragmentManager;
-    private UserFragment mUserFragment;
-    private AboutFragment mAboutFragment;
```

Comment:
With the exception of AboutFragment I think we want to make a request and reload the data each time we switch fragments.